src/main.c

```diff
@@ -194,24 +200,30 @@ void init_task(void *arg)
             &eth_sender_handle);
 
     configASSERT(status);
-
-
-
-    gpio.Mode = GPIO_MODE_OUTPUT_PP;
-    gpio.Pull = GPIO_NOPULL;
-    gpio.Pin = GPIO_PIN_13;
-    HAL_GPIO_Init(GPIOD, &gpio);
-    HAL_GPIO_WritePin(GPIOD, GPIO_PIN_13, GPIO_PIN_SET);
-
-    for (;;) {
-        if (!netif_is_link_up(netif)) {
+    uint8_t reset_state = 1;
+    for(;;){
+        if (!netif_is_link_up(netif))
+        {
             lcd_clear();
             lcd_print_string_at("Link:", 0, 0);
             lcd_print_string_at("down", 0, 1);
         }
-
-        HAL_GPIO_TogglePin(GPIOD, GPIO_PIN_13);
-        vTaskDelay(500);
+        uint16_t current_pin = choose_pin(current_mode);
+        // don't rest if it's already reset
+        if (current_pin == OFF_LEDS && reset_state)

```

Comment:
Please use more informative name for  "reset_state" - this might be for example "leds_turned_off" (in this case invert logic)